### PR TITLE
Feature: tag what region a NewGRF / Heightmap / Scenario is about

### PR DESCRIPTION
**Pull request by TrueBrain**
We make use of two Open Source lists: UN M49 and ISO 3166-2. These two together give ~5400 regions to indicate what your content is about.

### CODE REVIEW

File: bananas_api/tool_region/__main__.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            subdivision_name = subdivision_name.split("(")[0].strip()
+            subdivision_name = subdivision_name.split("[")[0].strip()
+
+            country_code = subdivision_code[:2]
+
+            regions.add((subdivision_code, subdivision_name, country_code))
+
+    cwd = os.path.dirname(__file__)
+
+    with open(f"{cwd}/../helpers/regions.py", "w") as fp:
```
Comment (LordAro) [inline bananas_api/tool_region/__main__.py:98]: I'm never a fan of scripts that decide where to output things for me. Perhaps stdout instead, with stderr for the INFO message?

Being able to just use `print()` would simplify having to do all those `\n`...

File: bananas_api/tool_region/__main__.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+
+@click_helper.command()
+@click.argument("m49_csv_file", type=click.Path(dir_okay=False, file_okay=True, exists=True))
+@click.argument("iso_3166_2_json_file", type=click.Path(dir_okay=False, file_okay=True, exists=True))
+def main(m49_csv_file, iso_3166_2_json_file):
+    regions = set()
+
+    with open(m49_csv_file) as fp:
+        for line in fp.readlines()[1:]:
```
Comment (LordAro) [inline bananas_api/tool_region/__main__.py:27]:
```suggestion
        next(fp)
        for line in fp:
```

? (possibly with a comment)

File: bananas_api/tool_region/__main__.py (hunk 1 above)
Comment (TrueBrain) [reply bananas_api/tool_region/__main__.py:98]: This is a lot of output, and the tool is designed to bring this to a very specific file. Important to observe it doesn't use your current working directory, but the location of the source file to find the correct location.

So I am not sure why you would prefer output to stdout, which means you have to pipe it yourself to this file? Feels like more work with no added benefit? Asking, as I might be missing something :)

Comment (LordAro) [reply bananas_api/tool_region/__main__.py:98]: Just personal preference. As the advantage of being able to preview output (or even diff it) before letting it trample over my working dir

Comment (TrueBrain) [reply bananas_api/tool_region/__main__.py:98]: The file is in git, so you can do all that! And that without accidentally getting 5000 lines of code over your stdout :)

Comment (TrueBrain) [reply bananas_api/tool_region/__main__.py:101]: After some fiddling around, this solution just sucks in general; and this is just a symptom.

I am going to add the CSV and JSON file to the repository, and load them on start-up. That way it is much more clear what is going on, etc.

File: bananas_api/helpers/regions.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
+import json
+import os
+import unicodedata
+
+REGIONS = {}
+
+folder = os.getcwd()
+if not os.path.exists(f"{folder}/region-un-m49.csv"):
```
Comment (frosch123) [inline bananas_api/helpers/regions.py:8]: This looks fragile to me, considering that PYTHONPATH is a list of paths.
I would have used
```python
folder = os.path.join(os.path.dirname(__file__), "..", "..")
```

Comment (TrueBrain) [reply bananas_api/helpers/regions.py:8]: Neither `PYTHONPATH` nor your suggestion is really any good. Your suggestion for example breaks if you `pip install -e .` the package.

As I mentioned in the other PR, I also don't like this file is copied from one repository to the other.

In result, most likely next weekend, I will spend some time moving this code away to its own library .. either in https://github.com/OpenTTD/py-helpers or its own (py-regions?). There I will spend the effort to actually proper package such files in a pip package, so it can be properly installed. That way we don't have to deal with this stuff, and that way we are sure both repositories use the same code. Sadly, packaging non-Python files in a pip package is not trivial, so I need to take some time to make that happen :)

Long story short, if you don't mind just going for "that-what-works" for now, I promise to make it pretty soon :)

**Review by frosch123 — APPROVED** [review]
